### PR TITLE
SOLR-13259: Add new section on Reindexing in Solr

### PR DESCRIPTION
**Pull request by ctargett**
While Solr frequently recommends to users that they reindex in various situations, we've had no central place in the Ref Guide that stated when they must or should do that, or how to do it.

This PR adds a new page `reindexing.adoc` under the section "Indexing and Basic Data Operations".

It also adds links to this new section in a few places (but probably not everywhere it could), and standardizes spelling on "reindex" instead of "re-index" throughout the guide.

### CODE REVIEW

File: solr/solr-ref-guide/src/reindexing.adoc [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+If you add or delete a field from Solr's schema, it's strongly recommended to reindex.
+
+When you add a field, you generally do so with the intent to use the field in some way.
+Since documents were indexed before the field was added, the index will not hold any references to the field for earlier documents.
+If you want to use the new field for faceting, for example, the new field facet will not include any documents that were not indexed with the new field.
+
+There is a slightly different situation when deleting a field.
+In this case, since simply removing the field from the schema doesn't change anything about the index, the field will still be in the index until the documents are reindexed.
+In fact, Lucene may keep a reference to a deleted field _forever_ (see also https://issues.apache.org/jira/browse/LUCENE-1761[LUCENE-1761]).
+This may only be an issue for your environment if you try to add a field that has the same name a deleted field,
```
Comment (gerlowskija) [inline solr/solr-ref-guide/src/reindexing.adoc]: "same name as a"

File: solr/solr-ref-guide/src/reindexing.adoc [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+There are several types of changes to Solr configuration that require you to reindex your data.
+
+These changes include editing properties of fields or field types; adding fields, field types, or copy field rules;
+upgrading Solr; and some system configuration properties.
+
+It's important to be aware that many changes require reindexing, because there are times when not reindexing
+can have negative consequences for Solr as a system, or for the ability of your users to find what they are looking for.
+
+There is no process in Solr for programmatically reindexing data. When we say "reindex", we mean, literally,
+"index it again". However you got the data into the index the first time, you will run that process again.
```
[the remark this answers is not in the document]
Comment (ctargett) [reply solr/solr-ref-guide/src/reindexing.adoc]: +1 thanks

[the remark this answers is not in the document]
Comment (ctargett) [reply solr/solr-ref-guide/src/reindexing.adoc]: changed it to "when the need arises"...if there's one thing this page should teach people is that they're going to need to reindex someday.

File: solr/solr-ref-guide/src/reindexing.adoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+Beyond specific field-level properties, <<analyzers.adoc#analyzers,analysis chains>> are also configured on field types, and are applied at index and/or query time.
+
+It's possible to define separate analysis chains for indexing and query events, or you can define a single chain
+that is applied to both event types.
+
+If you change the analysis chain that applies to indexing events, it is strongly recommended that you reindex.
+This is because all of the changes that occur due to the chain configuration are applied to documents as they are
+being indexed, and only reindexing will allow your changes to take effect on documents.
+
+While not reindexing after analyzer changes is not required, be aware that not reindexing can cause unexpected
```
Comment (tflobbe) [inline solr/solr-ref-guide/src/reindexing.adoc]: you mean "While reindexing..."?

File: solr/solr-ref-guide/src/reindexing.adoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+== Upgrades
+
+When upgrading between major versions (for example, from a 7.x release to 8.0 or 8.x), a best practice
+is to always reindex your data.
+The reason for this is that subtle changes may occur in default field type definitions or the underlying code.
+
+[NOTE]
+If you have *not* changed your schema as part of an upgrade from one minor release to another (such as, from 7.x
+to 7.x), you can often get away with reindexing your documents without first deleting your index. However, when
```
Comment (tflobbe) [inline solr/solr-ref-guide/src/reindexing.adoc]: `to 7.x+1`? Also, why is it recommended to upgrade for minor version upgrades?

File: solr/solr-ref-guide/src/reindexing.adoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+upgrading to a major release, you should plan to reindex your documents because of the likelihood of changes that break back-compatibility.
```
Comment (tflobbe) [inline solr/solr-ref-guide/src/reindexing.adoc]: This note is not clear to me. It seems to say that for minor version upgrades you can "reindex without deleting your index", but for major version upgrades "you should reindex". Is this second part supposed to be "reindex by first deleting your data/to a new collection"?

File: solr/solr-ref-guide/src/reindexing.adoc [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+With this approach, you will index your documents into a newly created collection and once everything is completed,
+create an alias for the collection and point your front-end at the collection alias. Queries will be routed
+to the new collection seamlessly.
+
+Here is an example of creating an alias that points to a single collection:
+
+[source,bash]
+http://localhost:8983/solr/admin/collections?action=CREATEALIAS&name=myData&collections=oldCollection
+
```
Comment (tflobbe) [inline solr/solr-ref-guide/src/reindexing.adoc]: Should it say than after that it's safe to delete the old collection using DELETE collections API? I suspect some people may be afraid of the DELETE acting on the alias and being explicit in the doc will save them some googling time

File: solr/solr-ref-guide/src/reindexing.adoc (hunk 3 above)
Comment (ctargett) [reply solr/solr-ref-guide/src/reindexing.adoc]: right!

File: solr/solr-ref-guide/src/reindexing.adoc (hunk 4 above)
Comment (ctargett) [reply solr/solr-ref-guide/src/reindexing.adoc]: Was supposed to be 'not upgrading'. Also made the versions more clear ('a later 7.x release')

File: solr/solr-ref-guide/src/reindexing.adoc (hunk 5 above)
Comment (ctargett) [reply solr/solr-ref-guide/src/reindexing.adoc]: I think I was making too subtle of a point - it's too awkwardly phrased to be clear. Rewording it to just say you usually don't have to reindex for minor version upgrades as long as you've made no changes that would usually trigger a need to reindex.

File: solr/solr-ref-guide/src/reindexing.adoc (hunk 6 above)
Comment (ctargett) [reply solr/solr-ref-guide/src/reindexing.adoc]: Good point, will add.

**Pull request merged**